### PR TITLE
Refactor service package into common application; add tool; update tools

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate Refactor common functionality from `service` package into generalized `application` package. This will allow us to create new "apps" that aren't necessarily services (like background workers). Refactor common functionality from existing tools into generalized `tool` package. Update all tools appropriately (except for one, deferred as tech debt to to effort involved).

Overall, this greatly simplifies the process of creating new services, tools, and other applications.

(Much of the previous work was leading up to this.)

### CODE REVIEW

File: tool/tool.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/tidepool-org/platform/log"
+)
+
+const (
+	HelpFlag    = "help"
+	VersionFlag = "version"
+	VerboseFlag = "verbose"
+)
+
+type Tool struct {
```
Comment (jh-bate) [inline tool/tool.go]: If its always a command line tool do you think the naming should reflect that?

File: tool/tool.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+func (t *Tool) Terminate() {
+	t.args = nil
+	t.cli = nil
+
+	t.Application.Terminate()
+}
+
+func (t *Tool) CLI() *cli.App {
```
Comment (jh-bate) [inline tool/tool.go]: hmmm so maybe this covers it off `theTool.CLI()` ?

**Comment by darinkrauss** [comment]
[[platform] Dexcom API - Prepare platform for new functionality](https://trello.com/c/I2mcmwqu/225-platform-dexcom-api-prepare-platform-for-new-functionality)

File: tool/tool.go (hunk 2 above)
Comment (darinkrauss) [reply tool/tool.go]: This is just access to the underlying third-party cli package.

File: tool/tool.go (hunk 1 above)
Comment (darinkrauss) [reply tool/tool.go]: It doesn't have to be. I expect that certain tools will be invoked from the admin site, once that gets implemented, via exec calls.

**Comment by darinkrauss** [comment]
[[platform] Download data from Dexcom API via OAuth](https://trello.com/c/MmE0hoUx/259-platform-download-data-from-dexcom-api-via-oauth)

**Pull request merged**